protocol/x/perpetuals/types/liquidity_tier.go

```diff
@@ -88,93 +99,43 @@ func (liquidityTier LiquidityTier) GetMaxAbsFundingClampPpm(clampFactorPpm uint3
-// - initialMarginQuoteQuantums = scaledInitialMarginPpm * quoteQuantums / 1_000_000
-//
-// note:
-// - divisions are delayed for precision purposes.
-func (liquidityTier LiquidityTier) GetInitialMarginQuoteQuantums(
-	bigQuoteQuantums *big.Int,
-	openInterestQuoteQuantums *big.Int,
+func (liquidityTier LiquidityTier) GetAdjustedInitialMarginPpm(
+	oiQuoteQuantums *big.Int,
 ) *big.Int {
-	bigOpenInterestUpperCap := new(big.Int).SetUint64(liquidityTier.OpenInterestUpperCap)
-
-	// If `open_interest` >= `open_interest_upper_cap` where `upper_cap` is non-zero,
-	// OIMF = 1.0 so return input quote quantums as the IMR.
-	if openInterestQuoteQuantums.Cmp(
-		bigOpenInterestUpperCap,
-	) >= 0 && liquidityTier.OpenInterestUpperCap != 0 {
-		return bigQuoteQuantums
-	}
+	oiCapUpper := lib.BigU(liquidityTier.OpenInterestUpperCap)
+	oiCapLower := lib.BigU(liquidityTier.OpenInterestLowerCap)
 
-	ratQuoteQuantums := new(big.Rat).SetInt(bigQuoteQuantums)
-
-	// If `open_interest_upper_cap` is 0, OIMF is disabled。
+	// If `open_interest_upper_cap` is 0, OIMF is disabled.
 	// Or if `current_interest` <= `open_interest_lower_cap`, IMF is not scaled.
-	// In both cases, use base IMF as OIMF.
-	bigOpenInterestLowerCap := new(big.Int).SetUint64(liquidityTier.OpenInterestLowerCap)
-	if liquidityTier.OpenInterestUpperCap == 0 || openInterestQuoteQuantums.Cmp(
-		bigOpenInterestLowerCap,
-	) <= 0 {
-		// Calculate base IMR: multiply `bigQuoteQuantums` with `initialMarginPpm` and divide by 1 million.
-		ratBaseIMR := lib.BigRatMulPpm(ratQuoteQuantums, liquidityTier.InitialMarginPpm)
-		return lib.BigRatRound(ratBaseIMR, true) // Round up initial margin.
+	if oiCapUpper.Sign() == 0 || oiQuoteQuantums.Cmp(oiCapLower) <= 0 {
+		return lib.BigU(liquidityTier.InitialMarginPpm)
 	}
 
-	// If `open_interest_lower_cap` < `open_interest` <= `open_interest_upper_cap`, calculate the scaled OIMF.
-	// `Scaling Factor = (Open Notional - Lower Cap) / (Upper Cap - Lower Cap)`
-	ratScalingFactor := new(big.Rat).SetFrac(
-		new(big.Int).Sub(
-			openInterestQuoteQuantums, // reuse pointer for memory efficiency
-			bigOpenInterestLowerCap,
-		),
-		bigOpenInterestUpperCap.Sub(
-			bigOpenInterestUpperCap, // reuse pointer for memory efficiency
-			bigOpenInterestLowerCap,
-		),
-	)
-
-	// `IMF Increase = Scaling Factor * (1 - Base IMF)`
-	ratIMFIncrease := lib.BigRatMulPpm(
-		ratScalingFactor,
-		lib.OneMillion-liquidityTier.InitialMarginPpm, // >= 0, since we check in `liquidityTier.Validate()`
-	)
-
-	// Calculate `Max(IMF Increase, 0)`.
-	if ratIMFIncrease.Sign() < 0 {
-		panic(
-			fmt.Sprintf(
-				"GetInitialMarginQuoteQuantums: IMF Increase is negative (%s), liquidityTier: %+v, openInterestQuoteQuantums: %s",
-				ratIMFIncrease.String(),
-				liquidityTier,
-				openInterestQuoteQuantums.String(),
-			),
-		)
+	// If `open_interest` >= `open_interest_upper_cap` where `upper_cap` is non-zero, OIMF is 1.
+	if oiCapUpper.Sign() > 0 && oiQuoteQuantums.Cmp(oiCapUpper) >= 0 {
```

Comment:
Yes I agree and was considering removing it, but I explicitly added it here in case code gets moved around later, or the check above gets removed or something